app/presenters/tree_node_builder.rb

```diff
@@ -186,7 +194,8 @@ def tooltip(tip)
   end
 
   def node_icon(icon)
-    ActionController::Base.helpers.image_path("100/#{icon}")
+    self.class.node_icon_cache[icon] ||=
```

Comment:
The mutex isn't actually preventing mutual writes to the cache.  It's just preventing it from being created more than once.